scipy/signal/_signaltools.py

```diff
@@ -1496,15 +1496,10 @@ def order_filter(a, domain, rank):
                              "should have an odd number of elements.")
 
     a = np.asarray(a)
-    if a.dtype in [object, 'float128']:
-        mesg = (f"Using order_filter with arrays of dtype {a.dtype} is "
-                f"deprecated in SciPy 1.11 and will be removed in SciPy 1.14")
-        warnings.warn(mesg, DeprecationWarning, stacklevel=2)
-
-        result = _sigtools._order_filterND(a, domain, rank)
-    else:
-        result = ndimage.rank_filter(a, rank, footprint=domain, mode='constant')
+    if a.dtype in [object, np.longdouble]:
+        raise ValueError(f"dtype={a.dtype} is not supported by order_filter")
```

Comment:
Can we use the same (kind) of check here as for `medfilt`? I.e. something like
```suggestion
    if not (np.issubdtype(volume.dtype, np.integer) 
            or volume.dtype in [np.float32, np.float64]):
        raise ValueError(f"dtype={volume.dtype} is not supported by order_filter")
```